src/main/java/bio/terra/common/stairway/StairwayComponent.java

```diff
@@ -142,4 +152,63 @@ private String getClusterName() {
     return String.format(
         "%s-%s", kubeService.getNamespace(), stairwayProperties.getClusterNameSuffix());
   }
+
+  public enum Status {
+    INITIALIZING,
+    OK,
+    ERROR,
+    SHUTDOWN,
+  }
+
+  /**
+   * Builder for providing inputs to the initialize method. This allows us to add parameters without
+   * adding new methods to the Stairway component.
+   */
+  public static class InitializerBuilder {
```

Comment:
optional nit: maybe `InitializationCustomization` or `StairwayCustomization`? While this is a builder class relating to the initialize function, `InitializerBuilder` isn't conveying a lot to me.

Reply:
I renamed it `StairwayOptionsBuilder`